platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java

```diff
@@ -108,20 +111,37 @@ public void setCrlLocation(String location) {
     /**
      * Generates a new CRL object from the given location.
      *
-     * @param location Path to the CRL file
-     * @return A CRL object constructed from the given file path. Null if an error occurred while attempting to read the file.
+     * @param location File Path or URL to the CRL file
+     * @return A CRL object constructed from the given file path or URL. Null if an error occurred while attempting to read the file.
      */
     private CRL createCrl(String location) {
-        try (FileInputStream fis = new FileInputStream(new File(location))) {
+        URL url = urlFromPath(location);
+
+        //If we get a URL, use URL, otherwise use as local file path
+        try (InputStream is = url != null ?
+                url.openStream() :
+                new FileInputStream(new File(location))) {
+
             CertificateFactory cf = CertificateFactory.getInstance("X.509");
-            return cf.generateCRL(fis);
+            CRL crl = cf.generateCRL(is);
+            is.close();
```

Comment:
You don't need this close here. The try with resources that you set up will close it automatically.